### PR TITLE
Stores the jwt token in a http cookie

### PR DESCRIPTION
**Pull request by pratap0007**
Adds a proxy to proxy the auth and rating endpoints
 - This proxy helps to configure the http cookies
in the client on ui side so that server can access it from .

Updates rating design to adds cookie and removes header 

Adds implementation to manage http cookie on server 
 - Adds endpoints for logout and get an accessToken
   Updates rating api and use manual validation of jwt which
   is fetched from cookie

Adds api endpoints in UI to fetch access token and logout
 - Adds `/user/accesstoken` endpoint on UI to get the access
   toke for user to show it on user profile
 - Adds `/user/logout` to delete the http cookie on server

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/setupProxy.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const {createProxyMiddleware} = require('http-proxy-middleware');
+
+module.exports = function (app) {
+  app.use(
+    createProxyMiddleware(['/auth', '/user'], {
+      target: 'http://localhost:4200',
```
Comment (PuneetPunamiya) [inline ui/src/setupProxy.js:6]: What will happen in case of production ?

Comment (PuneetPunamiya) [reply ui/src/setupProxy.js:6]: I think we can use the env value of `REDIRECT_URI` instead of hard coding this value

File: ui/src/setupProxy.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      changeOrigin: true
+    })
+  );
+  app.use(
+    createProxyMiddleware(['/resource/**/rating'], {
+      target: 'http://localhost:8000',
```
Comment (PuneetPunamiya) [inline ui/src/setupProxy.js:12]: same as above

File: api/pkg/auth/service/auth.go [hunk 3]
```diff
@@ -62,6 +62,20 @@ type Service interface {
 	HubAuthenticate(res http.ResponseWriter, req *http.Request)
 }
 
+type cookie struct {
```
Comment (PuneetPunamiya) [inline api/pkg/auth/service/auth.go:65]: Can you move this struct to [app.go](https://github.com/tektoncd/hub/blob/main/api/pkg/auth/app/app.go)

File: api/pkg/auth/service/auth.go [hunk 4]
```diff
@@ -192,6 +206,27 @@ func (s *service) HubAuthenticate(res http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	//Add cookie and response and send it to in header
+
```
Comment (PuneetPunamiya) [inline api/pkg/auth/service/auth.go:210]: nit: can you please delete this blank line

File: api/pkg/auth/service/auth.go [hunk 5]
```diff
@@ -229,3 +264,15 @@ func List(res http.ResponseWriter, req *http.Request) {
 		return
 	}
 }
+
+func (c cookie) createCookie(res http.ResponseWriter) {
```
Comment (PuneetPunamiya) [inline api/pkg/auth/service/auth.go:268]: Is it possible to add a unit test for this function

File: api/pkg/service/rating/rating.go [hunk 6]
```diff
@@ -50,10 +51,21 @@ func New(api app.Config) rating.Service {
...
+	newctx, err := s.JWTAuth(ctx, p.Session, requiredScope)
+	if err != nil {
+		return &rating.GetResult{Rating: 0}, err
+	}
+
 	req := request{
-		db:     s.DB(ctx),
-		log:    s.Logger(ctx),
-		userID: validator.UserID(ctx),
+		db:     s.DB(newctx),
```
Comment (PuneetPunamiya) [inline api/pkg/service/rating/rating.go:66]: Why can't we use the same context ?

File: api/pkg/service/rating/rating_http_test.go [hunk 7]
```diff
@@ -26,17 +26,16 @@ import (
...
 	"github.com/tektoncd/hub/api/gen/rating"
 	"github.com/tektoncd/hub/api/pkg/db/model"
-	"github.com/tektoncd/hub/api/pkg/service/validator"
 	"github.com/tektoncd/hub/api/pkg/testutils"
 	goa "goa.design/goa/v3/pkg"
 )
 
 func GetChecker(tc *testutils.TestConfig) *goahttpcheck.APIChecker {
-	service := validator.NewService(tc.APIConfig, "rating")
+	// service := validator.NewService(tc.APIConfig, "rating")
```
Comment (PuneetPunamiya) [inline api/pkg/service/rating/rating_http_test.go:34]: Do we need this anymore ?

File: api/pkg/service/rating/rating_http_test.go [hunk 8]
```diff
@@ -195,11 +194,11 @@ func TestGet_Http_ResourceNotFound(t *testing.T) {
 }
 
 func UpdateChecker(tc *testutils.TestConfig) *goahttpcheck.APIChecker {
-	service := validator.NewService(tc.APIConfig, "rating")
+	// service := validator.NewService(tc.APIConfig, "rating")
```
Comment (PuneetPunamiya) [inline api/pkg/service/rating/rating_http_test.go:197]: nit

File: ui/src/setupProxy.js (hunk 1 above)
Comment (pratap0007) [reply ui/src/setupProxy.js:6]: yes ,we should get it from env

File: api/pkg/service/rating/rating.go (hunk 6 above)
Comment (pratap0007) [reply api/pkg/service/rating/rating.go:66]: It was not adding userID in the context while doing jwtAuth validation

**Comment by sm43** [comment]
/approve

if I want to curl the rating api .. how can i pass the token now?

**Pull request closed without merging**